meson.build

```diff
@@ -129,32 +107,18 @@ prefixdir = get_option('prefix')
 if not prefixdir.startswith('/')
         error('Prefix is not absolute: "@0@"'.format(prefixdir))
 endif
-if prefixdir != rootprefixdir and rootprefixdir != '/' and not prefixdir.strip('/').startswith(rootprefixdir.strip('/') + '/')
-        error('Prefix is not below root prefix (now rootprefix=@0@ prefix=@1@)'.format(
-                rootprefixdir, prefixdir))
-endif
 
+prefixdir_noslash = prefixdir == '/usr/' ? '/usr' : prefixdir
```

Comment:
Hmm, that's a verrrry strict definition of "no slash". Maybe:
`prefixedir.rstrip('/') ?: '/'` ?

Reply:
This works:
```python
prefixdir_noslash = '/' + prefixdir.strip('/')
```

Reply:
`Unknown method "rstrip" in object <[OptionStringHolder] holds [OptionString]: '/usr'> of type OptionStringHolder.`

Reply:
> This works:
> 
> ```python
> prefixdir_noslash = '/' + prefixdir.strip('/')
> ```

doesn't that remove _all_ slashes? IE /some/prefix/ would become /someprefix

Reply:
It doesn't.

Reply:
What?

I'm so confused, the method is called strip() and not replace(). Why would it remove things from the middle?

Reply:
I guess I misread the documentation:

https://mesonbuild.com/Reference-manual_elementary_str.html#strstrip

> All characters in this string will be stripped.

as it says 'all', not 'all trailing'

Reply:
I would read that as a comparison of "all characters in this string" vs. "spaces and newlines".

But making the documentation clearer is always a free action... https://github.com/mesonbuild/meson/commit/1b8dcbcc5782a7809d23d5bff1baeb5fee74068e

(Oops, ignore the fact that the commit message says "split" instead of "strip".)

Reply:
thanks, that's nicer now